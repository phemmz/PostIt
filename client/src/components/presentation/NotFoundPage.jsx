import React from 'react';

/**
 * @description NotFoundPage
 * @return {*} div
 */
const NotFoundPage = () => {
  return (
    <div className="container">
      <div className="row">
        <div
          className="col s12 m6 offset-m3 card red-text custom center not-found"
        >
          <h4>
            <i
              className="fa fa-exclamation-triangle"
              aria-hidden="true"
            /> 404</h4>
          <h4>Oops!! Page Not Found</h4>
        </div>
      </div>
    </div>
  );
};

export default NotFoundPage;
