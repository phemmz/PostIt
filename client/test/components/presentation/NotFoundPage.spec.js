import React from "react";
import { mount } from "enzyme";
import NotFoundPage from '../../../src/components/presentation/NotFoundPage';

describe("NotFoundPage Component", () => {
  let mountedNotFoundPage;

  const notFoundPage = () => {
    if (!mountedNotFoundPage) {
      mountedNotFoundPage = mount(
        <NotFoundPage />
      );
    }
    return mountedNotFoundPage;
  }

  beforeEach(() => {
    mountedNotFoundPage: undefined;
  });

  it("should always render a div", () => {
    const divs = notFoundPage().find("div");
    expect(notFoundPage()).toMatchSnapshot();
    expect(divs.length).toBeGreaterThan(0);
  });

  it("should contain everything else that gets rendered", () => {
    const divs = notFoundPage().find("div");
    const wrappingDiv = divs.first();
    expect(wrappingDiv.children()).toEqual(notFoundPage().children());
  });
});

