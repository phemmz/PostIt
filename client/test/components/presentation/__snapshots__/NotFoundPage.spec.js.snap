// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage Component should always render a div 1`] = `
<NotFoundPage>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col s12 m6 offset-m3 card red-text custom center not-found"
      >
        <h4>
          <i
            aria-hidden="true"
            className="fa fa-exclamation-triangle"
          />
           404
        </h4>
        <h4>
          Oops!! Page Not Found
        </h4>
      </div>
    </div>
  </div>
</NotFoundPage>
`;
